presto-phoenix/src/main/java/io/prestosql/plugin/phoenix/PhoenixMetadata.java

```diff
@@ -391,4 +393,16 @@ private JdbcOutputTableHandle createTable(ConnectorSession session, ConnectorTab
             throw new PrestoException(PHOENIX_METADATA_ERROR, "Error creating Phoenix table", e);
         }
     }
+
+    @Override
+    public Optional<AggregationApplicationResult<ConnectorTableHandle>> applyAggregation(
+            ConnectorSession session,
+            ConnectorTableHandle table,
+            List<AggregateFunction> aggregates,
+            Map<String, ColumnHandle> assignments,
+            List<List<ColumnHandle>> groupBy)
+    {
+        // TODO support aggregation pushdown
+        return Optional.empty();
```

Comment:
shoudn't this be a default behaviour?

Reply:
it is. However, Phoenix extends JdbcMetadata, so inherits JdbcMetadata behavior, but it does not work for Phoenix case, as it does something special in `io.prestosql.plugin.phoenix.PhoenixClient#buildSql`.